File: data/readable_format/cpp/file1133.cpp
#include <string>
#include <iostream>
#include <cassert>

void stripPath(const std::string& fnam2, std::string& fnam, std::string& path, int& iflag) {
    iflag = 0;
    std::size_t pos = fnam2.rfind('/');
    if (pos != std::string::npos) {
        iflag = static_cast<int>(pos) + 1;
        fnam = fnam2.substr(pos + 1);
        path = fnam2.substr(0, pos+1);
    } else {
        fnam = fnam2;
        path = "";
    }
}

int main() {
    std::string fnam2, fnam, path;
    int iflag;

    // Test 1
    fnam2 = "/path/to/the/file.txt";
    stripPath(fnam2, fnam, path, iflag);
    assert(fnam == "file.txt");
    assert(path == "/path/to/the/");
    assert(iflag == 13);
    std::cout << "Test 1 passed." << std::endl;

    // Test 2
    fnam2 = "file.txt";
    stripPath(fnam2, fnam, path, iflag);
    assert(fnam == "file.txt");
    assert(path.empty());
    assert(iflag == 0);
    std::cout << "Test 2 passed." << std::endl;

    // Test 3
    fnam2 = "/path/to/multiple/slashes/file.txt";
    stripPath(fnam2, fnam, path, iflag);
    assert(fnam == "file.txt");
    assert(path == "/path/to/multiple/slashes/");
    assert(iflag == 26);
    std::cout << "Test 3 passed." << std::endl;

    return 0;
}

File: data/readable_format/fortran/file1133.f90
! Define the subroutine
subroutine strippath(fnam2,fnam,path,iflag)
  character(len=80) :: fnam2
  character(len=13) :: fnam
  character(len=80) :: path
  integer :: iflag, i
  
  iflag=0
  
  do i=1, len(fnam2)
    if(fnam2(i:i) .eq. '/') then
      iflag=i
    endif
  end do
  
  if(iflag.gt.0) then
    fnam=fnam2(iflag+1:)
    path=fnam2(1:iflag)
  else
    fnam=fnam2
    path=""
  endif
end subroutine strippath

! Unit tests
program testStripPath
  implicit none
  character(len=80) :: fnam2, path
  character(len=13) :: fnam
  integer :: iflag

  ! Test 1
  fnam2 = "/path/to/the/file.txt"
  call strippath(fnam2,fnam,path,iflag)
  print *, "Test 1 - Filename:", trim(fnam)
  print *, "Test 1 - Path:", trim(path)
  print *, "Test 1 - Position of last '/':", iflag

  ! Test 2
  fnam2 = "file.txt"
  call strippath(fnam2,fnam,path,iflag)
  print *, "Test 2 - Filename:", trim(fnam)
  print *, "Test 2 - Path:", trim(path)
  print *, "Test 2 - Position of last '/':", iflag

  ! Test 3
  fnam2 = "/path/to/multiple/slashes/file.txt"
  call strippath(fnam2,fnam,path,iflag)
  print *, "Test 3 - Filename:", trim(fnam)
  print *, "Test 3 - Path:", trim(path)
  print *, "Test 3 - Position of last '/':", iflag
end program testStripPath